gcovr/writer/json.py

```diff
@@ -112,64 +112,41 @@ def print_json_summary_report(covdata, output_file, options):
     )
 
     for key in keys:
-        (
-            filename,
-            line_total,
-            line_covered,
-            line_percent,
-            branch_total,
-            branch_covered,
-            branch_percent,
-            function_total,
-            function_covered,
-            function_percent,
-        ) = summarize_file_coverage(covdata[key], options.root_filter)
+        filename = presentable_filename(covdata[key].filename, options.root_filter)
 
         json_dict["files"].append(
             {
                 "filename": filename,
-                "line_total": line_total,
-                "line_covered": line_covered,
-                "line_percent": line_percent,
-                "branch_total": branch_total,
-                "branch_covered": branch_covered,
-                "branch_percent": branch_percent,
-                "function_total": function_total,
-                "function_covered": function_covered,
-                "function_percent": function_percent,
+                **_summary_from_stats(SummarizedStats.from_file(covdata[key]), None),
             }
         )
```

Comment:
What about adding optional `**kwargs` to `_summary_from_stats` which are added to the dict. Than he copy of the data for each file isn't needed anymore.

Reply:
I wouldn't expect that to affect performance noticeably. There is only going to be one copy per file in the output, which is very efficient compared to other parts of the code. This is Python, after all. I would totally agree with you if this were C++ or Rust, though :) 

I think flattening the return value from this function makes the data flow clearer than routing parts of the dict through the function.